src/SixLabors.Fonts/Buffer{T}.cs

```diff
@@ -26,21 +27,37 @@ public Buffer(int length)
             int bufferSizeInBytes = length * itemSizeBytes;
             this.buffer = ArrayPool<byte>.Shared.Rent(bufferSizeInBytes);
             this.length = length;
+
+            ByteMemoryManager<T> manager = new(this.buffer);
+            this.memory = manager.Memory.Slice(0, this.length);
+            this.span = this.memory.Span;
+
             this.isDisposed = false;
         }
 
+        /// <summary>
+        /// Gets an array slice over the buffer.
+        /// Do not allow this to escape the scope of the parent struct!!
+        /// </summary>
+        /// <returns>The <see cref="ArraySlice{T}"/>.</returns>
+        public ArraySlice<T> DangerousGetSlice()
+        {
+            if (this.buffer is null)
+            {
+                ThrowObjectDisposedException();
+            }
+
+            return new ArraySlice<T>(Unsafe.As<T[]>(this.buffer), 0, this.length);
```

Comment:
This is dangerous. Will look for a workaround.

Reply:
@antonfirsov Do you have any ideas here?

I never let anything access the buffer outside of the range, but it still doesn't feel right that I have this unsafe cast array that returns the length in the number of bytes.

Reply:
@antonfirsov sorry to bump. I'm just out of ideas without rewriting the interpreter to work with spans

Reply:
Decided to take the perf hit and go safe for now.